ci/free_disk_space.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+echo "=============================================================================="
+echo "Freeing up disk space on CI system"
+echo "=============================================================================="
+
+df -h
+echo "Removing unnecessary large directories"
+sudo rm -rf /usr/local/lib/android /usr/share/dotnet /usr/share/swift
```

Comment:
Why do we need to remove these in CI for a server?

Reply:
We were getting `disk storage full` errors when saving docker images, so we are deleting large directories that we don't need to free up space. This is an implementation inspired by Flink. https://github.com/apache/flink/blob/02d30ace69dc18555a5085eccf70ee884e73a16e/tools/azure-pipelines/free_disk_space.sh

Reply:
Does flink's Azure pipelines use the same image as GitHub Action? They seem to be two different CI env.
Maybe we need to use `du` to check the actual disk usage.
And let's add a comment for it.

Reply:
Not sure if they use the same Github Action but I manually _ssh-ed_ into the action runner to find out large directories.

Reply:
> And let's add a comment for it.

I'm sorry I did not understand. Comment for what? 😅

Reply:
> Maybe we need to use du to check the actual disk usage.

I think `df -h` gives us a better result.

Reply:
> > And let's add a comment for it.
> 
> I'm sorry I did not understand. Comment for what? 😅

Comment on why we decided to delete those directories. For example, we deleted the top three unnecessary large directories and they are found by command xxx...

Reply:
Understood! Just pushed my changes.

Reply:
Where can I find the new comment?

Reply:
[Here](https://github.com/apache/apisix/pull/8735/files#diff-54747011ef6df77ae48a9abd7cf3c62ee69410f22fa6a41cbf04e4852a6f6a1cR19-R22), in this [commit](https://github.com/apache/apisix/pull/8735/commits/e4d38d321ef46552e3121b65186c012094b36d87).

Reply:
Got it. Thanks!